Wrappers/Python/cil/processors/MaskGenerator.py

```diff
@@ -0,0 +1,338 @@
+                          You can secify only lower quantile value by setting another to None
+                          such as quantiles = (float,q2), then
+                          upper boundary will be amax(data). Similarly, to specify only upper 
+                          boundary, use quantiles = (None,q1). If both quantiles
+                          are set to None, then original array will be returned.
+      - mean              test element-wise if 
+                          abs(A - mean(A)) < threshold_factor * std(A).
+                          Default value of threshold_factor is 3. If no axis is specified, 
+                          then operates over flattened array. Alternatively operates along axis specified 
+                          as dimension_label.
+      - median            test element-wise if 
+                          abs(A - median(A)) < threshold_factor * scaled MAD(A),
+                          scaled median absolute deviation (MAD) is defined as 
+                          c*median(abs(A-median(A))) where c=-1/(sqrt(2)*erfcinv(3/2))
+                          Default value of threshold_factor is 3. If no axis is specified, 
+                          then operates over flattened array. Alternatively operates along axis specified 
+                          as dimension_label.
+      - movmean           the same as mean but uses rolling mean with a specified window,
+                          default window value is 5
+      - movmedian         the same as mean but uses rolling median with a specified window,
+                          default window value is 5
+
+      '''
+
+    def __init__(self,
+                 mode = 'special_values',
+                 threshold_value = (None, None),
+                 quantiles = (None, None),
+                 threshold_factor = 3,
+                 window = 5,
+                 axis = None):
+
+        kwargs = {'mode': mode,
+                  'threshold_value': threshold_value,
+                  'threshold_factor': threshold_factor,
+                  'quantiles': quantiles,
+                  'window': window,
+                  'axis': axis}
+
+        super(MaskGenerator, self).__init__(**kwargs)
+    
+    def check_input(self, data):
+        
+        if not (issubclass(type(data), DataContainer)):
+            raise TypeError('Processor supports only following data types:\n' +
+                            ' - ImageData\n - AcquisitionData\n' +
+                            ' - DataContainer')
+        if self.mode not in ['special_values', 'nan', 'inf', 'threshold', 'quantile',\
+                             'mean', 'median', 'movmean', 'movmedian']:
+            raise Exception("Wrong mode. One of the following is expected:\n" + \
+                            "special_values, nan, inf, threshold, \n quantile, mean, median, movmean, movmedian")
+        
+        if self.axis is not None:
```

Comment:
maybe `axis` could be a number and we'd accept additional containers without `dimension_label`

Reply:
can now accept integer values also